backend/dataall/modules/dataset_sharing/services/share_processors/s3_bucket_process_share.py

```diff
@@ -88,6 +89,8 @@ def process_approved_shares(
                 env_group,
             )
             try:
+                if not ShareObjectService.verify_principal_role(session, share):
```

Comment:
Minor. The exception message is a bit misleading. We can only revoke it has been approved. For users it is impossible to revoke this share

Reply:
addressed in [d216e9a](https://github.com/data-dot-all/dataall/pull/1161/commits/d216e9a3cc8e9024b759afadae86d6a20145582b)